core/server/data/importer/importers/data/posts.js

```diff
@@ -100,7 +100,13 @@ class PostsImporter extends BaseImporter {
 
                 // CASE: search through imported data.
                 // EDGE CASE: uppercase tag slug was imported and auto modified
-                let importedObject = _.find(this.requiredImportedData[tableName], {originalSlug: objectInFile.slug});
+                let importedObject = null;
+
+                if (objectInFile.slug) {
+                    importedObject = _.find(this.requiredImportedData[tableName], {originalSlug: objectInFile.slug});
+                } else {
+                    importedObject = _.find(this.requiredImportedData[tableName], {originalId: objectInFile.id});
```

Comment:
Do we need to check for the existence of `objectInFile.id`? 

I noticed above that the `originalIdMap` doesn't necessarily contain an entry for every id - just trying to work out if there's an edge we're missing here?

Reply:
The `objectInFile.id` is always there because of [this bit](https://github.com/TryGhost/Ghost/pull/10917/files#diff-f41262b74e0c309d6ef623f67e2dd79eR96). But you are right, making it `else if (objectInFile.id)` makes it proof to accidental ` === undefined` comparisons in case id assigning code changes in the future :+1:

Reply:
Aaaah I see! Yup that makes sense :relaxed: 

One more thing, what is the reason to attempt to match on slug first? Could we just always match on the id and remove the conditional?

Reply:
Very good point! Was thinking of slug matching as a fallback, but after rereading the previous statement there's no point in keeping it as the `originalId` will always be accessible :bulb:

Reply:
:dancer: !!